heaps/min_heap.py

```diff
@@ -21,18 +21,33 @@ def __init__(self):
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(1)
```

Comment:
👍  However the space complexity is O(log n) due to `heap_up` function's recursive call stack.